torch/csrc/Generator.cpp

```diff
@@ -79,6 +83,11 @@ static PyObject* THPGenerator_pynew(
         c10::DeviceTypeName(device.type()),
         " is not supported for torch.Generator() api.");
   }
+  self->weakreflist = NULL;
+
+  static py::handle _generator_registry = py::module::import("torch").attr("random").attr("_generator_registry");
```

Comment:
You should have a "release()" at the end here to make sure to leak the reference.